Automattic/tumblr-theme-garden
# handle npf and jsnpf tags with redvelvet

### Summary

Adds Redvelvet to convert Gutenberg post format into NPF for the NPF tag

+ uses Tumblr's js_string method for the JS modifier to ensure the exact format.

### Testing steps

- composer install
- From theme search, search and activate  "Streamer-Luck" theme
- Go to site and view source

you should see something like this:

```
posts.push({ npf: JSON.parse('{\x22content\x22:[{\x22type\x22:\x22text\x22,\x22text\x22:\x22This is a sticky post.\\n\\nThere are a few things to verify:\\n\\n\\tThe sticky post should be distinctly recognizable in some way in comparison to normal posts. You can style the .sticky class if you are using the post_class() function to generate your post classes, which is a best practice.\\n\\tThey should show at the very top of the blog index page, even though they could be several posts back chronologically.\\n\\tThey should still show up again in their chronologically correct postion in time, but without the sticky indicator.\\n\\tIf you have a plugin or widget that lists popular posts or comments, make sure that this sticky post is not always at the top of those lists unless it really is popular.\x22}],\x22layout\x22:[],\x22trail\x22:[],\x22version\x22:2}'), id: '1241'});
--
  |  
  | posts.push({ npf: JSON.parse('{\x22content\x22:[{\x22type\x22:\x22text\x22,\x22text\x22:\x22This is an audio block\x22},{\x22type\x22:\x22text\x22,\x22text\x22:\x22\x22},{\x22type\x22:\x22audio\x22,\x22url\x22:\x22http:\\/\\/localhost:8881\\/wp-content\\/uploads\\/2024\\/12\\/sample_with_thumbnail.mp3\x22,\x22title\x22:\x22my title\x22,\x22artist\x22:\x22my artist\x22,\x22album\x22:\x22my album\x22,\x22poster\x22:[{\x22url\x22:\x22\x22}]},{\x22type\x22:\x22text\x22,\x22text\x22:\x22\x22},{\x22type\x22:\x22text\x22,\x22text\x22:\x22\x22}],\x22layout\x22:[],\x22trail\x22:[],\x22version\x22:2}'), id: '2045'});
  |  
  | posts.push({ npf: JSON.parse('{\x22content\x22:[{\x22type\x22:\x22audio\x22,\x22url\x22:\x22http:\\/\\/localhost:8881\\/wp-content\\/uploads\\/2024\\/12\\/sample_with_thumbnail.mp3\x22,\x22title\x22:\x22title 1\x22,\x22artist\x22:\x22artist 1\x22,\x22album\x22:\x22album 1\x22,\x22poster\x22:[{\x22url\x22:\x22http:\\/\\/localhost:8881\\/wp-content\\/uploads\\/2024\\/10\\/cropped-cropped-apple.png\x22}]}],\x22layout\x22:[],\x22trail\x22:[],\x22version\x22:2}'), id: '2041'});
  |  
  | posts.push({ npf: JSON.parse('{\x22content\x22:[{\x22type\x22:\x22audio\x22,\x22url\x22:\x22http:\\/\\/localhost:8881\\/wp-content\\/uploads\\/2024\\/11\\/101audio.mp3\x22,\x22title\x22:\x22aaaa\x22,\x22artist\x22:\x22bbb\x22,\x22album\x22:\x22ccc\x22,\x22poster\x22:[{\x22url\x22:\x22http:\\/\\/localhost:8881\\/wp-content\\/uploads\\/2024\\/10\\/cropped-cropped-apple.png\x22}]}],\x22layout\x22:[],\x22trail\x22:[],\x22version\x22:2}'), id: '2036'});
  |  
  | posts.push({ npf: JSON.parse('{\x22content\x22:[{\x22type\x22:\x22text\x22,\x22text\x22:\x22hmm: says me\x22},{\x22type\x22:\x22text\x22,\x22text\x22:\x22\x22},{\x22type\x22:\x22text\x22,\x22text\x22:\x22hmm2: says: you\x22},{\x22type\x22:\x22text\x22,\x22text\x22:\x22\x22},{\x22type\x22:\x22text\x22,\x22text\x22:\x22\x22}],\x22layout\x22:[],\x22trail\x22:[],\x22version\x22:2}'), id: '1837'});
  |  
  | posts.push({ npf: JSON.parse('{\x22content\x22:[{\x22type\x22:\x22text\x22,\x22text\x22:\x22someone: hello world\\nsomeone else: hmm\\nok: what?\x22}],\x22layout\x22:[],\x22trail\x22:[],\x22version\x22:2}'), id: '1845'});
  |  
  | posts.push({ npf: JSON.parse('{\x22content\x22:[{\x22type\x22:\x22text\x22,\x22text\x22:\x22hello?\x22}],\x22layout\x22:[],\x22trail\x22:[],\x22version\x22:2}'), id: '1834'});
  |  
  | posts.push({ npf: JSON.parse('{\x22content\x22:[{\x22type\x22:\x22text\x22,\x22text\x22:\x22dsdsd\x22}],\x22layout\x22:[],\x22trail\x22:[],\x22version\x22:2}'), id: '1846'});
  |  
  | posts.push({ npf: JSON.parse('{\x22content\x22:[{\x22type\x22:\x22text\x22,\x22text\x22:\x22Welcome to WordPress. This is your first post. Edit or delete it, then start writing!\x22}],\x22layout\x22:[],\x22trail\x22:[],\x22version\x22:2}'), id: '1'});
  |  
  | posts.push({ npf: JSON.parse('{\x22content\x22:[{\x22type\x22:\x22text\x22,\x22text\x22:\x22alert(\\\x22hi\\\x22);\x22}],\x22layout\x22:[],\x22trail\x22:[],\x22version\x22:2}'), id: '1847'});
  |  
  | posts.push({ npf: JSON.parse('{\x22content\x22:[{\x22type\x22:\x22text\x22,\x22text\x22:\x22hmm\x22}],\x22layout\x22:[],\x22trail\x22:[],\x22version\x22:2}'), id: '1848'});
```

### Reviewing tips

🧁 Request a review to team CupcakeLabs to assign a random team member.

🔄 Feel free to ping the team name again to "re-roll" or add another team member.

🚀 For now there is no requirement on getting review approvals before merging.
